UnitsNet/CustomCode/Wrappers/Pressure.Wrapper.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnitsNet.Units;
+using UnitsNet.CustomCode.Units;
+
+namespace UnitsNet.CustomCode.Wrappers
+{
+    /// <summary>
+    /// From gratestas in #422
+    /// Pressure is a state function, for its measurement depends on environmental factors such as ambient pressure, elevation above sea level and local weather conditions. For two persons located in the different environment to speak of the same pressure, one must relate it to a reference. There are three basis reference: absolute, gauge, vacuum
+    ///    Absolute is zero-referenced to the total vacuum.
+    ///    Gauge refers to a level of the local atmospheric pressure.
+    ///    Vacuum is the negative of the gauge.
+    ///Therefore, to obtain consistent and qualitative data, the reference measurement is crucial.
+    /// </summary>
+    public struct ReferencePressure
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="pressure"></param>
+        /// <param name="reference"></param>
+        public ReferencePressure(Pressure pressure, PressureReference reference)
+        {
+            _reference = reference;
+            Pressure = pressure;
+        }
+
+        /// <summary>
+        /// ctor using BaseReference of absolute to assign default _reference
+        /// </summary>
+        /// <param name="pressure"></param>
+        public ReferencePressure(Pressure pressure)
+        {
+            _reference = BaseReference;
+            Pressure = pressure;
+        }
+
+        /// <summary>
+        ///     The public repersentation of the measured reference this quantity was constructed with.
+        /// </summary>
+        public PressureReference Reference => _reference.GetValueOrDefault(BaseReference);
+
+        /// <summary>
+        ///     The measured reference this quantity was constructed with.
+        /// </summary>
+        private readonly PressureReference? _reference;
+
+        /// <summary>
+        /// List property of reference options: Gauge, Absolute, and Vacuum
+        /// </summary>
+        public static List<PressureReference> References { get; } = Enum.GetValues(typeof(PressureReference)).Cast<PressureReference>().Except(new[] { PressureReference.Undefined }).ToList();
+
+        /// <summary>
+        ///     The base reference representation of this quantity for the numeric value stored internally. All conversions go via this value.
+        /// </summary>
+        public static PressureReference BaseReference { get; } = PressureReference.Absolute;
+
+        private Pressure Pressure { get; }
```

Comment:
This is a matter of preference, but I personally favor private readonly fields over private getter properties. It seems more typical C#.